### ISSUE TITLE
Refactor changes to 'vmsyscall' and merge into master

### ISSUE THREAD

**Issue by connorkuehl** [enarx/enarx#371]
Chiefly, instead of `FrameRange` we could have a `Frame` newtype and `FrameRange` could be replaced with `core::ops::Range` or `enarx/span::Span`. Implementing `PartialOrd` and `Ord` on these new types will simplify some of the ordering done to the memory `Region`s.

The fact that `core::ops::Range` is not `Copy` may complicate things. This change will also require changes to `enarx-keep-sev` and possibly `enarx-keep-sev-shim`.

See #370, this could should be on the staging branch once that is merged.

**Comment by connorkuehl** [comment]
Actually, with the new shape everything is taking, I think it makes more sense to track the progress of the keep and shim in their respective meta issues.
